temp/static_analysis.md

```diff
@@ -0,0 +1,549 @@
+Or print-out the DataObject in a dataframe style,
+with an index on the main supporting mesh entity id
+```pycon
+DPF DataObject:
+Nodal stress S1 [on *named_selection_name*] # if present
+ node ID   step  component   value
+       1      1         S1     0.1
+       2      1         S1     0.2
+       3      1         S1     0.3
+       ...
+       1      2         S1     0.2
+       2      2         S1     0.3
+       3      2         S1     NaN
+```
+
+Get the name of the DataObject
+print(stress_S1.name)
+Nodal stress S1
+
+Get the mesh support of the DataObject
+print(stress_S1.mesh)
+
+Get the shape of the DataObject
+```pycon
+>>> print(stress_S1.shape)
+list of lengths of each dimension present: [nb_steps, nb_entities,
+```
+
+Get specific data from a DataObject (as a new DataObject)
+```pycon
+>>> stress_S1_step_1 = stress_S1[1]  # Equivalent to stress_S1[0, ...] or stress_S1[0, :, :]
+```
+or
+```pycon
+>>> stress_S1_step_1 = stress_S1.step(1)
+```
+
+```pycon
+>>> print(stress_S1_step_1)
+Nodal stress S1
+For:
+- steps: 1
+- named selection: {named_selection[0]}
+```
+
+### Get the minimum, maximum, mean of the DataObject 
+Use DPF operators for efficiency.
+
+Across all dimensions present in the specific DataObject, unless specified as argument
+Take inspiration from the "axis" argument for ``numpy.ndarray.amax``.
+Do we want a ``nanmax``? Which propagates NaN values? (Whether we have actual NaNs or not)
+These would return DataObjects or a scalar.
+```pycon
+>>> global_max_stress_S1 = stress_S1.max()
+```
+```pycon
+>>> max_stress_S1_step_1 = stress_S1_step_1.max()
+```
+or
+```pycon
+max_stress_S1_step_1 = stress_S1.max(steps=[1])
+```
+
+Arguments would include lists of steps, nodes, elements, components... a named selection
+```pycon
+>>> max_stress_S1_node_1 = stress_S1.max(nodes=[1])
+```
+
+Get the mesh support for a specific DataObject (uses the scoping associated to the DataObject)
+```pycon
+>>> mesh_1 = max_stress_S1_node_1.mesh
+>>> print(mesh_1)
+Mesh containing:
+- 1 nodes: [1]
+- 0 elements: []
+```
+
+### Plot a DataObject (3D contour plots)
+TODO:
+plotting and graphing functions can return the Plotter instance
+
+Plot a multistep DataObject - takes an optional "step" argument, defaults to last
+```pycon
+>>> pl = stress_S1.plot(step=3, return_plotter=True)
+```
+
+Plot a one-step DataObject - the "step" argument is not read
+```pycon
+>>> stress_S1_step_1.plot()
+```
+
+Plot principal stresses
+
+TODO: Not really sure how this is different from plotting results
+
+
+### Create graphs (2D curve plots)
+TODO:
+the graph method would take optional "x", "y", "names" arguments.
+
+Graph a multistep DataObject
+```pycon
+>>> max_stress_S1_node_1.graph(x="steps", names="ID")
+```
+or as proposed in the GitHub discussion #213
+```pycon
+>>> max_stress_S1_node_1.plot(graph=True, individual_figures=False)
+```
+
+Plot waterfall/stagger diagrams -> a waterfall diagram is different from a waterfall plot!
+
+Plot waterfall plots
```

Comment:
I think the difference is that the way you wrote it, the filtering is done at the extraction step, which is of course a possibility but a specific extraction should not be required before every operation.
We will need the possibility to work on only part of a DataObject via filtering after extraction.
Your proposed `.iterate()` method is one way of solving part of this issue, yet I think we will need `.plot` methods for our objects, with DataObject-specific arguments.
In your example, the data is filtered and treated so it can be used by the matplotlib plot function. This is one way of doing it but IMHO our goal is to provide helpers so that users can do all this using our APIs.